File: src/dataset/Java/proverb_exercism-uzilan/gemini/GeminiFlash_Proverb_v3.java
class Proverb {

    private final String[] words;

    Proverb(final String[] words) {
        this.words = words;
    }

    String recite() {
        if (words.length == 0) {
            return "";
        }

        final StringBuilder sb = new StringBuilder();
        int n = words.length;
        for (int i = 0; i < n - 1; i++) {
            sb.append(String.format("For want of a %s the %s was lost.\n", words[i], words[i + 1]));
        }
        sb.append(String.format("And all for the want of a %s.", words[0]));
        return sb.toString();
    }
}

File: src/dataset/javascript/proverb_exercism-ThomasZumsteg/gemini/GeminiFlash_proverb_v3.js
class Proverb {
    constructor(...args) {
        this.words = [];
        this.qualifier = undefined;

        for (const arg of args) {
            if (typeof arg === 'string') {
                this.words.push(arg);
            } else if (typeof arg === 'object' && arg !== null && arg.qualifier !== undefined) {
                this.qualifier = arg.qualifier;
            }
        }
    }

    toString() {
        const result = [];
        const wordsLength = this.words.length;

        for (let i = 1; i < wordsLength; i++) {
            result.push(`For want of a ${this.words[i - 1]} the ${this.words[i]} was lost.`);
        }

        const lastWord = this.words[0];
        const qualifierPrefix = this.qualifier ? `${this.qualifier} ` : '';
        result.push(`And all for the want of a ${qualifierPrefix}${lastWord}.`);

        return result.join('\n');
    }
}

module.exports = Proverb;